fus.c

```diff
@@ -867,16 +867,61 @@ resolve_all_solvables (Pool  *pool,
   return solv_failed;
 }
 
+/**
+ * remove_from_selection:
+ * @pool: initialized pool
+ * @selection: a selection of solvables
+ * @solvid: Id of solvable to be subtracted from the selection
+ */
+static void
+remove_from_selection (Pool  *pool,
+                       Queue *selection,
+                       Id     solvid)
+{
+    g_auto(Queue) remove;
+    queue_init (&remove);
+    selection_make (pool, &remove, pool_solvid2str (pool, solvid), SELECTION_CANON);
+    selection_subtract (pool, selection, &remove);
```

Comment:
You can replace this `selection_subtract` call by a `SELECTION_SUBTRACT` flag in `selection_make`. You don't even have to create the Queue.

Reply:
Which means this function is not even needed.